openrr-nav-viewer/examples/dwa_gui.rs

```diff
@@ -234,12 +234,16 @@ fn main() {
                         &result,
                         [current_pose.translation.x, current_pose.translation.y],
                     );
+                    let len = result.len();
                     let mut locked_angle_table = cloned_nav.angle_table.lock();
                     locked_angle_table
                         .insert(ROTATION_COST_NAME.to_owned(), current_pose.rotation.angle());
                     match nearest_path_point {
-                        Some(p) => {
-                            locked_angle_table.insert(PATH_DIRECTION_COST_NAME.to_owned(), p.1[2]);
+                        Some((idx, _)) => {
+                            locked_angle_table.insert(
+                                PATH_DIRECTION_COST_NAME.to_owned(),
+                                result[(idx + 20).min(len - 1)][2],
+                            );
```

Comment:
I think it's better to define some values to explain what you are doing here.

Below is an example, do not merge in the UI.

```suggestion
                            const FORWARD_OFFSET: usize = 20;
                            let look_ahead_idx = (idx + FORWARD_OFFSTE).min(len - 1);
                            locked_angle_table.insert(
                                PATH_DIRECTION_COST_NAME.to_owned(),
                                result[look_ahead_idx][2],
                            );
```